docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
On private keys, public_key() is a method rather than a property, and I think that's because it actually creates a new public key object when it is called. I would think we'd want ECDH to be consistent with that if we kept the name the same. Even private_numbers() is a method.

As for compute_key(), I'm good with that if you prefer it to compute_shared() or compute_shared_key().